code/game/objects/items/grenades/clusterbuster.dm

```diff
@@ -86,25 +90,26 @@
 
 /obj/item/slime_extract/proc/activate_slime()
 	var/list/slime_chems = src.activate_reagents
-	for(var/i in 1 to slime_chems.len)
-		if(!QDELETED(src))
-			var/chem = pick_n_take(slime_chems)
-			var/amount = 5
-			if(chem == "lesser plasma") //In the rare case we get another rainbow.
-				chem = "plasma"
-				amount = 4
-			if(chem == "holy water and uranium")
-				chem = "uranium"
-				reagents.add_reagent("holywater")
-			reagents.add_reagent(chem,amount) //Add them in random order so we get all effects
+	if(!QDELETED(src))
+		var/chem = pick(slime_chems)
+		var/amount = 5
+		if(chem == "lesser plasma") //In the rare case we get another rainbow.
+			chem = "plasma"
+			amount = 4
+		if(chem == "holy water and uranium")
+			chem = "uranium"
+			reagents.add_reagent("holywater")
+		reagents.add_reagent(chem,amount) //Add them in random order so we get all effects
```

Comment:
change the comment please, we no longer add them in a random order, we simply pick a random one that works